src/logic/function/basic.lean

```diff
@@ -466,6 +467,12 @@ lemma eq_update_iff {a : α} {b : β a} {f g : Π a, β a} :
   g = update f a b ↔ g a = b ∧ ∀ x ≠ a, g x = f x :=
 funext_iff.trans $ forall_update_iff _ (λ x y, g x = y)
 
+@[simp] lemma update_eq_self_iff : update f a b = f ↔ b = f a := by simp [update_eq_iff]
+@[simp] lemma eq_update_self_iff : f = update f a b ↔ f a = b := by simp [eq_update_iff]
+
+lemma ne_update_self_iff : f ≠ update f a b ↔ f a ≠ b := eq_update_self_iff.not
+lemma update_ne_self_iff : update f a b ≠ f ↔ b ≠ f a := update_eq_self_iff.not
+
```

Comment:
`logic.function.basic` is about to gain a "synchronized with mathlib4" label, so this will require a matching PR to mathlib4.

Reply:
mathlib4#558

Reply:
Awesome! Our first synchronization PR, I think. :-)